workflow/controller/operator.go

```diff
@@ -108,12 +108,12 @@ func newWorkflowOperationCtx(wf *wfv1.Workflow, wfc *WorkflowController) *wfOper
 // and its pods and decides how to proceed down the execution path.
 // TODO: an error returned by this method should result in requeuing the workflow to be retried at a
 // later time
-func (woc *wfOperationCtx) operate() {
+func (woc *wfOperationCtx) operate(wfdx sqldb.DBRepository) {
```

Comment:
Should the repository be a part of the context? What happens if you call `operate` with different databases?

Reply:
DBRepository is an interface type.  you can implement with any DB (RDBMS, NOSQL and Memcache).  operate function will take all DB implementation

Reply:
I do not quite understand what would be the behavior of 
```
woc.operate(db1)
...
woc.operate(db2)
...
woc.operate(nil)
...
woc.operate(db1)
```

Reply:
I started this feature to support centralized and multi-tenancy persistence storage. But as per current requirement and user feedback, I reduced the scope of this feature to support only centralized persistent in v2.3.  context is holding persistence session for centralized storage (in future we can implement map to hold the sessions for each namespace and common). The appropriate session will be passed to woc.operate(<>) to persist the workflow (current implementation will pass centralized session to all workflow.

Here examble:

 woc.operate(db1) --> Workflow will be persisted in DB1
...
woc.operate(db2) --> Workflow will be persisted in DB1
...
woc.operate(nil)  --> Persistence is not enabled, it will fallback to current storage (ETCD)
...
woc.operate(db1)  --> Workflow will be persisted in DB1